### PR TITLE
Start working on stage2 ARM backend

### PR DESCRIPTION
**Pull request by joachimschmidt557**
Added `codgen/arm.zig` with some basic functionality (load/store, data processing, branching, software interrupts)

I'm creating this as a draft pull request so people can observe the progress and incrementally provide feedback.

ToDo:

- [x] genBreakpoint
- [x] genCall
- [x] ret
- [x] genAsm
- [x] genSetReg
- [x] Hello World test

### CODE REVIEW

**Comment by andrewrk** [comment]
btw in case you haven't noticed, there is a rich amount of target info available to you in codegen.zig, especially for ARM.

`Function.bin_file.base.options.target.cpu` is [this struct](https://github.com/ziglang/zig/blob/b0846b6ecbb3e2557c5c95ddee04ecc055881d75/lib/std/target.zig#L482) and that `features` field is a set of [these target features](https://github.com/ziglang/zig/blob/master/lib/std/target/arm.zig)

In particular the "sub-architecture" features may be particularly important, e.g. `v8a` or `v4t`.

**Comment by joachimschmidt557** [comment]
@andrewrk Thanks! I wasn't aware of that.

**Comment by joachimschmidt557** [comment]
I think that I will focus this PR on getting Hello World up and running first using a (I think) common subset of the ARM instruction set. Afterwards, I'll include more instructions etc. and make the codegen code actually adapt to the feature set of the CPU.

File: src-self-hosted/codegen.zig [hunk 1]
```diff
@@ -75,7 +75,7 @@ pub fn generateSymbol(
             switch (bin_file.base.options.target.cpu.arch) {
                 .wasm32 => unreachable, // has its own code path
                 .wasm64 => unreachable, // has its own code path
-                //.arm => return Function(.arm).generateSymbol(bin_file, src, typed_value, code, dbg_line, dbg_info, dbg_info_type_relocs),
+                .arm => return Function(.arm).generateSymbol(bin_file, src, typed_value, code, dbg_line, dbg_info, dbg_info_type_relocs),
                 //.armeb => return Function(.armeb).generateSymbol(bin_file, src, typed_value, code, dbg_line, dbg_info, dbg_info_type_relocs),
```
Comment (andrewrk) [inline src-self-hosted/codegen.zig:79]: I don't suppose you could do armeb at the same time? Seems like it should be pretty simple, just use the correct endianness when writing integers into the code array

Comment (joachimschmidt557) [reply src-self-hosted/codegen.zig:79]: Ok, I'll do that as well.

**Comment by andrewrk** [comment]
Great!

**Pull request merged**